packages/entity-server/src/routes/hierarchy/middleware/attachCommonContext.ts

```diff
@@ -2,17 +2,19 @@
  * Tupaia
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
-import { NextFunction, Response } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import { PermissionsError } from '@tupaia/utils';
-import { SingleEntityRequest, MultiEntityRequest } from '../types';
 import { extractFieldsFromQuery, extractFieldFromQuery } from './fields';
+import { CommonContext } from '../types';
 
 const throwNoAccessError = (hierarchyName: string) => {
   throw new PermissionsError(`No access to requested hierarchy: ${hierarchyName}`);
 };
 
 export const attachCommonContext = async (
-  req: SingleEntityRequest | MultiEntityRequest,
+  req: Request<{ hierarchyName: string }, any, any, { field?: string; fields?: string }> & {
+    ctx: CommonContext;
+  },
```

Comment:
Had to make these middleware types a bit less strict since now we are using them for a whole gambit of different `...Request` types. However I'm actually glad to do this. I think the general rule for function signatures should be 'be as relaxed as is safe with the types you accept'. That leads to the highest level of flexibility and polymorphism 👍